### PR TITLE
Rename SyntheticNode to InnerNode, wrap it in a new SyntheticNode that has an inbound message queue

### PR DESCRIPTION
**Pull request by lwojtow**
1. Extract InnerNode
2. Add channels for InnerNode -> SyntheticNode messages
3. Basic setup for tests

### CODE REVIEW

File: src/tools/inner_node.rs [hunk 1]
```diff
@@ -67,21 +76,24 @@ impl SyntheticNode {
...
             crypto,
             tls: Tls {
                 acceptor,
                 connector,
             },
-        }
+        };
+        node.enable_handshake().await;
+        node.enable_reading().await;
+        node.enable_writing().await;
```
Comment (ljedrz) [inline src/tools/inner_node.rs:95]: I'd move those to `SyntheticNode::new`

File: src/tools/synth_node.rs [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+};
+
+pub struct SyntheticNode {
+    inner: InnerNode,
+    receiver: Receiver<(SocketAddr, BinaryMessage)>,
+}
+
+impl SyntheticNode {
+    pub async fn new(config: pea2pea::Config) -> Self {
+        let (sender, receiver) = mpsc::channel(100);
```
Comment (ljedrz) [inline src/tools/synth_node.rs:24]: this one should probably be a `const` too

File: src/tools/inner_node.rs (hunk 1 above)
Comment (lwojtow) [reply src/tools/inner_node.rs]: Done

File: src/tools/synth_node.rs (hunk 2 above)
Comment (lwojtow) [reply src/tools/synth_node.rs]: Done

File: src/tools/constants.rs [hunk 3]
```diff
@@ -0,0 +1,7 @@
+use std::time::Duration;
+
+/// Timeout for [SyntheticNode] when waiting for expected message.
+pub const EXPECTED_MESSAGE_TIMEOUT: Duration = Duration::from_secs(20);
+
+/// Channel buffer bound for [InnerNode] -> [SyntheticNode] messages.
+pub const SYNTH_NODE_BUFFER_SIZE: usize = 100;
```
Comment (ljedrz) [inline src/tools/constants.rs:7]:
```suggestion
pub const SYNTH_NODE_QUEUE_DEPTH: usize = 100;
```

Comment (lwojtow) [reply src/tools/constants.rs]: Done

**Review by ljedrz — APPROVED** [review]
LGTM :+1:

**Pull request merged**